cadcutils/cadcutils/net/groups_client.py

```diff
@@ -0,0 +1,587 @@
+#
+#         url = "{0}/users/{1}?idType={2}".format(self.base_url,
+#         identity.name, identity.type)
+#         xml_string = self._download_xml(url)
+#         reader = UserReader()
+#         user = reader.read(xml_string)
+#         self.logger.info('Retrieved user {0}'.format(user.internal_id))
+#         return user
+
+
+def print_group(group):
+    print('Group:')
+    print('             ID: {}'.format(group.group_id))
+    print('    Description: {}'.format(group.description))
+    print('          Owner: {}'.format(group.owner.identities['X500'].name))
+    print('  Last Modified: {}'.format(group.last_modified))
+    print('    User Admins: {}'.format((',\n'+' '*17).join(
+        ['{} ({})'.format(
+            a.identities['HTTP'].name,
+            a.identities['X500'].name) for a in group.user_admins])))
+    print('   Group Admins: {}'.format(
+        (',\n'+' '*17).join([gr.group_id for gr in group.group_admins])))
+    print_group_members(group)
+
+
+def print_group_members(group):
+    print('   User Members: {}'.format((',\n'+' '*17).join(
+        ['{} ({})'.format(
+            m.identities['HTTP'].name,
+            m.identities['X500'].name) for m in group.user_members])))
+    print('  Group Members: {}'.format(
+        (',\n'+' '*17).join([gr.group_id for gr in group.group_members])))
+
+
+def main_app():
+    parser = util.get_base_parser(version=version.version, auth_required=True,
+                                  default_resource_id=CADC_AC_SERVICE_ID)
+
+    parser.description = (
+        'Client for access the User Group Management System (GMS) at the '
+        'Canadian Astronomy Data Centre')
+
+    subparsers = parser.add_subparsers(
+        dest='cmd',
+        help='supported commands. Use the -h|--help argument of a command '
+             'for more details')
+
+    list_parser = subparsers.add_parser(
+        'list',
+        description='Retrieve name of all the groups the user has access to',
+        help='Retrieve name of all the groups the user has access to')
+    list_parser.add_argument(
+        '-r', '--role', choices=['admin', 'member', 'owner'],
+        help='Filter by role in the group',
+        required=False)
+    list_parser.epilog = (
+        'Example:\n'
+        '        cadc-groups list --cert ~/.ssl/cadcproxy.pem\n')
+
+    get_parser = subparsers.add_parser(
+        'get',
+        description='Retrieve information regarding CADC groups',
+        help='Retrieve information regarding CADC groups')
+    get_parser.add_argument(
+        'groupid', help='the id of the CADC group', nargs='+')
+    get_parser.epilog = (
+        'Example:\n'
+        '        cadc-groups get --cert $HOME/.ssl/cadcproxy.pem FOO\n')
+
+    create_parser = subparsers.add_parser(
+        'create',
+        description='Creates a new CADC Group. Caller is owner and member of'
+                    'the new group',
+        help='Creates a new CADC Group. Caller is owner and member of the new '
```

Comment:
Add information about which symbols are allowed in group ids?

Reply:
Good point.